crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -294,18 +294,84 @@ impl Mesh {
         self.attributes.insert(name.into(), values);
     }
 
-    /// Retrieve the data currently set behind a vertex attribute.
+    /// Retrieves an immutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute(&self, name: impl Into<Cow<'static, str>>) -> Option<&VertexAttributeValues> {
         self.attributes.get(&name.into())
     }
 
+    /// Retrieves a mutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute_mut(
         &mut self,
         name: impl Into<Cow<'static, str>>,
     ) -> Option<&mut VertexAttributeValues> {
         self.attributes.get_mut(&name.into())
     }
 
+    /// Retrieves multiple mutable references to _distinct_ vertex attributes.
+    /// Returns None at index i if the mesh had no data for the vertex attribute given at index i.
+    ///
+    /// # Panics
+    /// Panics if any of the given attribute names resolve to the same attribute, as this would otherwise allow aliased mutability.
```

Comment:
I'm not super familiar with the rendering side of things: I take it there's a reason we can't just supply a `HashSet`? Generally I prefer type-enforced rather than panic-enforced rules of this sort.

At first glance I'm guessing this is impossible because a) performance costs (arrays are GPU-friendly) and b) it looks like you can have synonyms that point to the same data.

Reply:
It's not impossible and I did consider a HashSet. However I think the const array approach provides a nice interface as you can immediately destructure the array into the same amount of attribute you provided. I think this is much more readable than getting a HashSet and then having to get keys out of this.

Besides, I think most use cases of this function will be just a static array where the distinctness of the attributes is obvious, and a HashSet would just be a tedious intermediary. I think users would often just assume the keys exist and panic themselves anyway.

The better performance is just gravy on top :)